classes/install/Upgrade.inc.php

```diff
@@ -1903,6 +1903,53 @@ function moveReviewerFiles() {
 			$result->MoveNext();
 		}
 		$result->Close();
+
+	}
+
+	/**
+	 * Move 'date_published' from the table 'published_submissions' into the table 'submission_settings'
+	 * @return boolean
+	 */
+	function migrateDatesPublished() {
+		$submissionDao = Application::getSubmissionDAO();
+		$result = $submissionDao->retrieve('SELECT submission_id, date_published FROM published_submissions');
+
+		while (!$result->EOF) {
+			$row = $result->GetRowAssoc(false);
+			$submissionId = $row['submission_id'];
+			$datePublished = $submissionDao->datetimeFromDB($row['date_published']);
+
+			$submission = $submissionDao->getById($submissionId);
+			$submission->setDatePublished($datePublished);
+			$submissionDao->updateLocaleFields($submission);
```

Comment:
If you need the information before the DB XML schemas are executed and thus the columns removed, you can create a temporary table in a preupdate script, s. for example commentsToEditor migration and dbscripts/xml/upgrade/3.0.0_preupdate_commentsToEditor.xml.